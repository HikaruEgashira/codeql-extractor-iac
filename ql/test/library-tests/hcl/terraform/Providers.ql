private import codeql.hcl.Terraform
private import codeql.iac.Dependencies

query predicate files(Terraform::TerraformFile f) { any() }

query predicate terraformSettings(Terraform::Terraform t) { any() }

query predicate requiredProvider(Terraform::RequiredProvider p) { any() }
