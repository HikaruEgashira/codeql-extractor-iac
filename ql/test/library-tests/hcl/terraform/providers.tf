# We strongly recommend using the required_providers block to set the
# Azure Provider source and version being used
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "=3.0.0"
    }
    time   = "~>0.7.2"
    random = "~>3.3.1"
  }
}

# Configure the Microsoft Azure Provider
provider "azurerm" {
  skip_provider_registration = true
  features {}
}
