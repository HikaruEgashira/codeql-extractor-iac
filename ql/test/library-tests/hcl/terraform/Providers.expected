files
| providers.tf:0:0:0:0 | providers.tf |
terraformSettings
| providers.tf:3:1:10:1 | terraform |
requiredProvider
| providers.tf:5:15:8:5 | RequiredProvider |
dependencies
| providers.tf:5:15:8:5 | hashicorp/azurerm@3.0.0 |
